common/subdevice-model.cpp

```diff
@@ -1749,10 +1749,13 @@ namespace rs2
         std::string fw_version = s->supports(RS2_CAMERA_INFO_FIRMWARE_VERSION) ?
             s->get_info(RS2_CAMERA_INFO_FIRMWARE_VERSION) : "";
 
-        bool supported_fw = s->supports(RS2_CAMERA_INFO_FIRMWARE_VERSION) ?
```

Comment:
regarding this method, had some comment from Nir, and it now looks like:

bool subdevice_model::supports_on_chip_calib()
    {
        bool is_d400 = s->supports(RS2_CAMERA_INFO_PRODUCT_LINE) ?
            std::string(s->get_info(RS2_CAMERA_INFO_PRODUCT_LINE)) == "D400" : false;
        std::string fw_version = s->supports(RS2_CAMERA_INFO_FIRMWARE_VERSION) ?
            s->get_info(RS2_CAMERA_INFO_FIRMWARE_VERSION) : "";
        bool supported_fw = s->supports(RS2_CAMERA_INFO_FIRMWARE_VERSION) ?
            is_upgradeable("05.11.12.0", fw_version) : false;
        bool d400_on_chip_calib_supported = s->is<rs2::depth_sensor>() && is_d400 && supported_fw;

        bool is_d500 = s->supports(RS2_CAMERA_INFO_PRODUCT_LINE) ?
            std::string(s->get_info(RS2_CAMERA_INFO_PRODUCT_LINE)) == "D500" : false;
        bool d500_on_chip_calib_supported = s->is<rs2::depth_sensor>() && is_d500;

        return d400_on_chip_calib_supported || d500_on_chip_calib_supported;
        // TODO: Once auto-calib makes it into the API, switch to querying camera info
    }

Reply:
Done